sdk/servicebus/service-bus/test/stress/stressTestsBase.ts

```diff
@@ -360,105 +365,92 @@ export class SBStressTestsBase {
-              `Number of successful closes for receiver so far: ${this.closeInfo.receiver.numberOfSuccesses}`,
-              `Number of failed closes for receiver so far: ${this.closeInfo.receiver.numberOfFailures}`
-            ]
-          : []
-      )
-      .reduce((output, entry) => output + "\n" + entry)
-      .concat("\n\n");
-    
-    await appendFile(this.reportFileName, currentSnapshot);
-    defaultClient.flush();
 
-    console.log(currentSnapshot);
-  }
+    eventProperties["elapsedTimeInSeconds"] = elapsedTimeInSeconds;
+    eventProperties["messsages.sent"] = this.messagesSent.length;
+    eventProperties["messages.received"] = this.messagesReceived.length;
 
-  public async end() {
+    if (this.snapshotOptions.snapshotFocus.includes("send-info")) {
+      eventProperties["send.pass"] = this.sendInfo.numberOfSuccesses;
+      eventProperties["send.fail"] = this.sendInfo.numberOfFailures;
+    }
 
-    defaultClient.flush();
+    if (this.snapshotOptions.snapshotFocus.includes("receive-info")) {
+      eventProperties["receive.pass"] = this.receiveInfo.numberOfSuccesses;
+      eventProperties["receive.fail"] = this.receiveInfo.numberOfFailures;
+    }
+
+    if (this.snapshotOptions.snapshotFocus.includes("message-lock-renewal-info")) {
+      eventProperties["lockRenewal.pass"] = this.messageLockRenewalInfo.numberOfSuccesses;
+      eventProperties["lockRenewal.fail"] = this.messageLockRenewalInfo.numberOfFailures;
+    }
+
+    if (this.snapshotOptions.snapshotFocus.includes("session-lock-renewal-info")) {
+      eventProperties["sessionLockRenewal.pass"] = this.sessionLockRenewalInfo.numberOfSuccesses;
+      eventProperties["sessionLockRenewal.fail"] = this.sessionLockRenewalInfo.numberOfFailures;
+    }
+
+    if (this.snapshotOptions.snapshotFocus.includes("close-info")) {
+      eventProperties["close.sender.pass"] = -this.closeInfo.sender.numberOfSuccesses;
+      eventProperties["close.sender.fail"] = this.closeInfo.sender.numberOfFailures;
+      eventProperties["close.receiver.pass"] = this.closeInfo.receiver.numberOfSuccesses;
+      eventProperties["close.receiver.fail"] = this.closeInfo.receiver.numberOfFailures;
+    }
 
-    // Logging all the errors to a file
     const errors = [].concat(
       this.sendInfo.errors,
       this.receiveInfo.errors,
       this.messageLockRenewalInfo.errors,
       this.sessionLockRenewalInfo.errors
     );
-    if (this.snapshotOptions.snapshotFocus.includes("receive-info")) {
-      await saveDiscrepanciesFromTrackedMessages(
-        this.trackedMessageIds,
-        this.messagesReportFileName
-      );
-    }
-    if (errors.length) {
-      await writeFile(
-        `errors-logged-${this.queueName}.txt`,
-        errors.reduce((output, entry) => output + "\n" + entry)
-      );
+
+    // TODO: it would be nicer to report the errors as they occur rather than
+    // doing this big dump of errors at the end.
```

Comment:
Can this todo be removed now?

Reply:
Updated. The issue is still there, but it's not about doing it only at the end - it's more about wanting to report in real-time.